wasi/testdata/args.go

```diff
@@ -0,0 +1,10 @@
+package main
+
+import (
+	"fmt"
+	"os"
+)
+
+func main() {
```

Comment:
I know you insist on using `main` to test this, please comment why, and if we expect args to never be able to be called except from main

Reply:
Ex. if this were me, and I had read the spec and noticed something like XXX must be able to be called from _start. I would make that the comment and link it.

If there was also a spec saying XXX must never be able to be called except from _start, I would note that and also link it.

Anything that's personal interpretation, ex. something says should not.. are our choices and should also be documented. Ex if we choose not to test, expect from `main`/`_start`, and the spec doesn't indicate that (ex it says MAY SHOULD NOT or anything besides MUST or MUST NOT), we can say either we choose to not allow something (and test it isn't allowed) or comment that we aren't testing from another place as it is undefined.

Reply:
Thanks for the feedback. In fact, I don't insist on using `main`, to be precise calling all functions from `main`, at all. Actually I once wrote the test so that the test go code calls exported functions outside `_start`. The comment `// Let TinyGo runtime initialize the WASI environment by calling _start` you commented before was an old comment I wrote at that time, and it's now back.
So, I've updated so that the test code calls exported functions after calling `_start` only once. I've also commented why it calls `_start` and also stated that the WASI spec's restriction doesn't affect wazero side.